File: sign.py
from flask import request
from bd_acess import *


def sign_up():
    firstName = request.form['firstName']
    lastName = request.form['lastName']
    username = request.form['username']
    email = request.form['email']
    phoneNumber = request.form['number']
    password = request.form['password']
    passwordConfirmation = request.form['passwordConfirmation']
    gender = request.form['gender']
    userData = {"firstName": firstName, "lastName": lastName, "username": username, "email": email,
                "phoneNumber": phoneNumber, "password": password, "passwordConfirmation": passwordConfirmation, "gender": gender}
    sign_up_response = write_in_db(
        firstName, lastName, username, email, phoneNumber, password, passwordConfirmation, gender)
    return userData, sign_up_response


def sign_in():
    user = request.form.get('ent')
    password = request.form.get('password-login')
    sign_in_response = read_from_db(user, password)

    return sign_in_response


if __name__ == '__main__':
    print(sign_up())

File: front_end/js/sign.js
const sign_in = document.getElementById("sign-in");
const sign_up = document.getElementById("sign-up");

sign_in.addEventListener("click", event => {
  if (sign_in.classList.contains("changeColor")) { }
  else { sign_in.classList.add("changeColor") };
  if (sign_up.classList.contains("changeColor")) sign_up.classList.remove("changeColor");
})

sign_up.addEventListener("click", event => {
  if (sign_up.classList.contains("changeColor")) { }
  else { sign_up.classList.add("changeColor") };
  if (sign_in.classList.contains("changeColor")) sign_in.classList.remove("changeColor")
})


// cadastro

const urlCadastro = "http://localhost:3000/sign-up"

const cadastrar = document.getElementById("send")

cadastrar?.addEventListener("click", e => {
  e.preventDefault();

  const firstName = document.getElementById("firstName").value;
  const lastName = document.getElementById("lastName").value;
  const username = document.getElementById("username").value;
  const email = document.getElementById("email").value;
  const password = document.getElementById("password").value;
  const passwordConfirmation = document.getElementById("passwordConfirmation").value;
  const number = document.getElementById("number").value;
  const gender = document.querySelector("input[name='gender']").value;

  if (password === passwordConfirmation) {
    const data = {
      firstName,
      lastName,
      username,
      email,
      password,
      number,
      gender
    }
    axios.post(urlCadastro, data)
      .then(response => {
        console.log(response.data)
      })
      .catch(error => console.log(error.response.data));
  }
  else {
    alert("As senha estão diferentes!")
  }

})

//login
const urlLogin = "http://localhost:3000/sign-in"
const logar = document.getElementById("enviar")

logar?.addEventListener("click", e => {
  e.preventDefault();

  const ent = document.getElementById("ent").value;
  const passwordLogin = document.getElementById("passwordLogin").value;

  const data = { ent, passwordLogin };

  axios.post(urlLogin, data)
    .then(response => {
      console.log(response.data)
    })
    .catch(error => console.log(error.response.data));

})


















